symengine/rational.h

```diff
@@ -20,8 +20,17 @@ class Rational : public Number
 
 public:
     IMPLEMENT_TYPEID(RATIONAL)
-    //! Constructor of Rational class
+//! Constructor of Rational class
+#if SYMENGINE_INTEGER_CLASS == SYMENGINE_BOOSTMP
```

Comment:
meant to ask about that. The only constructor for `SymEngine::Rational` is `Rational(rational_class i)`.  Likewise `SymEngine::Integer` has only the constructor `Integer(integer_class i)`.  Because neither parameter is passed by reference, copies are made in both cases.  The `Rational` constructor then initializes the private member `i` using `i{i}`, again copying `i`.  The `Integer` constructor initializes the private member `i` with `i{std::move(i)}`, avoiding the second copy.  The difference strikes me as purposeful, but I don't understand the reason.

It seems more logical to me to have two constructors for each class, much like copy and move constructors, but from rational_class and integer_class respectively:
`Rational(const rational_class &_i)`
`Rational(rational_class &&_i)`
`Integer(const integer_class &_i)`
`Integer(integer_class &&_i)`